pkg/tbtc/signing_loop.go

```diff
@@ -316,64 +324,50 @@ func (srl *signingRetryLoop) start(
 				srl.attemptCounter,
 			)
 
-			latestEndBlock, err := srl.doneCheck.exchange(
+			err = srl.doneCheck.signalDone(
```

Comment:
Just making sure about the intention. Currently:
1. A participating member just broadcasts the produced signature through the done message and actually abandons it for a while
2. Then, it relies on the incoming done messages, including the message from itself, in order to extract the final signature that should be the same in all received done messages. 

If this is what we want, I think we are good. I was just wondering about the possible security implications of the fact that the given member does not compare incoming signatures against the one produced on its own. However, I don't see any problems at the first sight.

Reply:
Yes, this is the intention. At first, I refactored the `signingDoneCheck` to be a bit more stateful and capture the signature, but the current approach looks cleaner in the code: everyone has to send, and all signatures must match. It's also important to note we accept only one message from the given sender.